dune-release distrib: unknown revision on Windows

Hi,

When running dune-release on Windows, I'm getting the following error:

```
$ dune-release distrib
[-] Building source archive
dune-release.exe: [ERROR] Exit code 128 from command
    `git --git-dir .git show -s --format=%ct 0.16.0^{commit}`:
  fatal: ambiguous argument '0.16.0^commit': unknown revision or path not in the
 working tree.
  Use '--' to separate paths from revisions, like this:
  'git <command> [<revision>...] -- [<file>...]'
```

But running the command directly works:

```
$ git --git-dir .git show -s --format=%ct 0.16.0^{commit}
1605539115
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -57,6 +57,7 @@
   distrib` (#325, @lehy).  Otherwise uncommitted changes to
   dune-project would be silently ignored by `dune-release distrib`.
 - Fix rewriting of github references in changelog (#330, @gpetiot)
+- Fixes a bug under cygwin where dune-release was unable to find the commit hash corresponding to the release tag (#329, @gpetiot)
 
 ### Security
 

--- a/lib/vcs.ml
+++ b/lib/vcs.ml
@@ -97,12 +97,12 @@ let dirtify_if ~dirty r id =
 
 let git_commit_id ~dirty r commit_ish =
   let dirty = dirty && commit_ish = "HEAD" in
-  let id = Cmd.(v "rev-parse" % "--verify" % (commit_ish ^ "^{commit}")) in
+  let id = Cmd.(v "rev-parse" % "--verify" % (commit_ish ^ "^0")) in
   run_git_string ~dry_run:false r id ~default:Default.string >>= fun id ->
   dirtify_if ~dirty r id
 
 let git_commit_ptime_s ~dry_run r commit_ish =
-  let commit_ish = commit_ish ^ "^{commit}" in
+  let commit_ish = commit_ish ^ "^0" in
   let time = Cmd.(v "show" % "-s" % "--format=%ct" % commit_ish) in
   run_git_string ~dry_run ~force:true r time ~default:Default.string
   >>= fun ptime ->

--- a/tests/bin/errors/run.t
+++ b/tests/bin/errors/run.t
@@ -46,7 +46,7 @@ Let's make `dune-release` run a `git`-command that's doomed to fail. After the c
     [-] Using tag "0.1.0"
     dune-release: [ERROR] Due to invalid commit-ish `1`:
       Exit code 128 from command
-        `git --git-dir .git rev-parse --verify 1^{commit}`:
+        `git --git-dir .git rev-parse --verify 1^0`:
       fatal: Needed a single revision
     
     [3]

--- a/tests/bin/non-github-doc-uri/run.t
+++ b/tests/bin/non-github-doc-uri/run.t
@@ -47,7 +47,7 @@ We do the whole dune-release process
     [-] Building source archive
     => rmdir _build/whatever-0.1.0.build
     -: exec: git --git-dir .git rev-parse --verify 0.1.0
-    => exec: git --git-dir .git show -s --format=%ct 0.1.0^{commit}
+    => exec: git --git-dir .git show -s --format=%ct 0.1.0^0
     => exec: git --git-dir .git clone --local .git _build/whatever-0.1.0.build
     => exec:
          git --git-dir _build/whatever-0.1.0.build/.git --work-tree   _build/whatever-0.1.0.build/ checkout --quiet -b dune-release-dist-0.1.0   0.1.0

--- a/tests/bin/non-github-uri/run.t
+++ b/tests/bin/non-github-uri/run.t
@@ -47,7 +47,7 @@ We do the whole dune-release process
     [-] Building source archive
     => rmdir _build/whatever-0.1.0.build
     -: exec: git --git-dir .git rev-parse --verify 0.1.0
-    => exec: git --git-dir .git show -s --format=%ct 0.1.0^{commit}
+    => exec: git --git-dir .git show -s --format=%ct 0.1.0^0
     => exec: git --git-dir .git clone --local .git _build/whatever-0.1.0.build
     => exec:
          git --git-dir _build/whatever-0.1.0.build/.git --work-tree   _build/whatever-0.1.0.build/ checkout --quiet -b dune-release-dist-0.1.0   0.1.0

--- a/tests/bin/url-file/run.t
+++ b/tests/bin/url-file/run.t
@@ -47,7 +47,7 @@ We make a dry-run release:
     [-] Building source archive
     => rmdir _build/whatever-0.1.0.build
     -: exec: git --git-dir .git rev-parse --verify 0.1.0
-    => exec: git --git-dir .git show -s --format=%ct 0.1.0^{commit}
+    => exec: git --git-dir .git show -s --format=%ct 0.1.0^0
     => exec: git --git-dir .git clone --local .git _build/whatever-0.1.0.build
     => exec:
          git --git-dir _build/whatever-0.1.0.build/.git --work-tree   _build/whatever-0.1.0.build/ checkout --quiet -b dune-release-dist-0.1.0   0.1.0

--- a/tests/bin/x-commit-hash/run.t
+++ b/tests/bin/x-commit-hash/run.t
@@ -46,7 +46,7 @@ We make a dry-run release
     [-] Building source archive
     => rmdir _build/whatever-0.1.0.build
     -: exec: git --git-dir .git rev-parse --verify 0.1.0
-    => exec: git --git-dir .git show -s --format=%ct 0.1.0^{commit}
+    => exec: git --git-dir .git show -s --format=%ct 0.1.0^0
     => exec: git --git-dir .git clone --local .git _build/whatever-0.1.0.build
     => exec:
          git --git-dir _build/whatever-0.1.0.build/.git --work-tree   _build/whatever-0.1.0.build/ checkout --quiet -b dune-release-dist-0.1.0   0.1.0

